OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/AdminControllerTests.cs

```diff
@@ -0,0 +1,780 @@
+        {
+            Entities = applications,
+            TotalAmount = applications.Count,
+        });
+
+        // Act
+        var result = await controller.GetApplications(new ApplicationFilter()).ConfigureAwait(false) as OkObjectResult;
+
+        // Assert
+        result.Should().NotBeNull();
+        result.StatusCode.Should().Be(StatusCodes.Status200OK);
+    }
+
+    [Test]
+    public async Task GetProviders_WhenNoRecordsInDB_ReturnsNoContentResult()
+    {
+        // Arrange
+        sensitiveProviderService.Setup(x => x.GetByFilter(It.IsAny<ProviderFilter>()))
+            .ReturnsAsync(new SearchResult<ProviderDto> { TotalAmount = 0, Entities = new List<ProviderDto>() });
+
+        currentUserService.Setup(x => x.IsAdmin()).Returns(true);
+
+        // Act
+        var result = await controller.GetByFilterProvider(new ProviderFilter()).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<NoContentResult>(result);
+    }
+
+    [Test]
+    [TestCase(0)]
+    public void Delete_WhenIdIsInvalid_ReturnsBadRequestObjectResult(long id)
+    {
+        // Arrange
+        sensitiveDirectionService.Setup(x => x.Delete(id));
+
+        // Act and Assert
+        Assert.ThrowsAsync<ArgumentOutOfRangeException>(
+            async () => await controller.DeleteDirectionById(id).ConfigureAwait(false));
+    }
+
+    [Test]
+    public void GetApplications_WhenCalledParentOrProvider_ShouldThrowUnauthorizedAccess()
+    {
+        // Arrange
+        sensitiveApplicationService.Setup(s => s.GetAll(It.IsAny<ApplicationFilter>())).ThrowsAsync(new UnauthorizedAccessException());
+
+        // Act & Assert
+        Assert.ThrowsAsync<UnauthorizedAccessException>(async () => await controller.GetApplications(new ApplicationFilter()));
+    }
+
+    [Test]
+    [TestCase(10)]
+    public async Task Delete_WhenIdIsInvalid_ReturnsNull(long id)
+    {
+        // Arrange
+        sensitiveDirectionService.Setup(x => x.Delete(id)).ReturnsAsync(Result<DirectionDto>.Success(direction));
+
+        // Act
+        var result = await controller.DeleteDirectionById(id) as OkObjectResult;
+
+        // Assert
+        Assert.That(result, Is.Null);
+    }
+
+    [Test]
+    public async Task GetApplications_WhenCollectionIsEmpty_ShouldReturnNoContent()
+    {
+        // Arrange
+        sensitiveApplicationService.Setup(s => s.GetAll(It.IsAny<ApplicationFilter>())).ReturnsAsync(new SearchResult<ApplicationDto>()
+        {
+            Entities = new List<ApplicationDto>(),
+            TotalAmount = 0,
+        });
+
+        // Act
+        var result = await controller.GetApplications(new ApplicationFilter()).ConfigureAwait(false) as NoContentResult;
+
+        // Assert
+        result.Should().NotBeNull();
+        result.StatusCode.Should().Be(StatusCodes.Status204NoContent);
+    }
+
+    [Test]
+    public async Task GetMinistryAdmins_WhenCalled_ReturnsOkResultObject_WithExpectedCollectionDtos()
```

Comment:
(Test naming) Test name starts with `GetMinistryAdmins_` while method it tests is `GetByFilterMinistryAdmin()`. Please avoid such ambiguity in this and subsequent tests.